client/app/components/Filters.jsx

```diff
@@ -0,0 +1,122 @@
+  multiple: PropTypes.bool,
+  current: PropTypes.oneOfType([
+    PropTypes.any,
+    PropTypes.arrayOf(PropTypes.any),
+  ]).isRequired,
+  values: PropTypes.arrayOf(PropTypes.any).isRequired,
+});
+
+export const FiltersType = PropTypes.arrayOf(FilterType);
+
+function createFilterChangeHandler(filters, onChange) {
+  return (filter, value) => {
+    if (filter.multiple && includes(value, ALL_VALUES)) {
+      value = [...filter.values];
+    }
+    if (filter.multiple && includes(value, NONE_VALUES)) {
+      value = [];
+    }
+    const allFilters = map(filters, f => (f.name === filter.name ? { ...filter, current: value } : f));
+    const changedFilter = find(allFilters, f => f.name === filter.name);
+    onChange(allFilters, changedFilter);
+  };
+}
+
+export function filterData(rows, filters = []) {
+  if (!isArray(rows)) {
+    return [];
+  }
+
+  let result = rows;
+
+  if (isArray(filters) && (filters.length > 0)) {
+    // "every" field's value should match "some" of corresponding filter's values
+    result = result.filter(row => every(
+      filters,
+      (filter) => {
+        const rowValue = row[filter.name];
+        const filterValues = isArray(filter.current) ? filter.current : [filter.current];
+        return some(filterValues, (filterValue) => {
+          if (moment.isMoment(rowValue)) {
+            return rowValue.isSame(filterValue);
+          }
+          // We compare with either the value or the String representation of the value,
+          // because Select2 casts true/false to "true"/"false".
+          return (filterValue === rowValue) || (String(rowValue) === filterValue);
+        });
+      },
+    ));
+  }
+
+  return result;
+}
+
+export function Filters({ filters, onChange }) {
+  if (filters.length === 0) {
+    return null;
+  }
+
+  onChange = createFilterChangeHandler(filters, onChange);
+
+  return (
+    <div className="filters-wrapper">
+      <div className="parameter-container container bg-white">
+        <div className="row">
+          {map(filters, (filter) => {
+            const options = map(filter.values, value => (
+              <Select.Option key={value}>{value}</Select.Option>
+            ));
+
+            return (
+              <div key={filter.name} className="col-sm-6 p-l-0 filter-container">
+                <label>{filter.friendlyName}</label>
+                <Select
+                  className="w-100"
+                  mode={filter.multiple ? 'multiple' : 'default'}
+                  value={filter.current}
+                  allowClear={filter.multiple}
+                  onChange={value => onChange(filter, value)}
+                >
+                  {!filter.multiple && options}
+                  {filter.multiple && [
+                    <Select.Option key={NONE_VALUES}><i className="fa fa-square-o m-r-5" />Clear</Select.Option>,
+                    <Select.Option key={ALL_VALUES}><i className="fa fa-check-square-o m-r-5" />Select All</Select.Option>,
```

Comment:
@kravets-levko the selectbox check/uncheck is a nice indication, but I think confusing - it looks like one is selected and the other deselected.

I know this is probably out of scope, but these really should be buttons and better have them at the top-right corner of the input and not inside it.
Use case - I clicked "Select all", got tons selected, then had to click on an empty space in the multi-select box to reach the "clear" action. Not so good.

Reply:
(Oh I see there's an "allowClear" but still not best ux)

Reply:
@ranbena I decided to keep existing behavior of this component (and even improved it a little bit), but I think it will be better to change something here after this PR merged, as a separate feature.

Reply:
I agree that fixing the UX issue is out of scope, but let's remove the icons. Like @ranbena, I found them confusing.